### ISSUE TITLE
[Bug]: Puppeteer Generated PDF is not same as Chrome Generated PDF

### ISSUE THREAD

**Issue by zahidhasanemon** [puppeteer/puppeteer#9228]
### Bug description

Puppeteer PDF is generating pdf (from url) different from chrome. In a single page pdf, chrome pdf is okay and generating a single page pdf.
[chrome_single_page.pdf](https://github.com/puppeteer/puppeteer/files/9958897/chrome_single_page.pdf)
But puppeteer is adding a blank page at the end.
[puppeteer_single_page.pdf](https://github.com/puppeteer/puppeteer/files/9958903/puppeteer_single_page.pdf)

When I have long content and multiple page, chrome is generating the pdf okay.
[chrome_multi_page.pdf](https://github.com/puppeteer/puppeteer/files/9958907/chrome_multi_page.pdf)
Puppeteer is however adding an extra blank page at the start and content spreads into multiple pages.
 
[puppeteer_multi_page.pdf](https://github.com/puppeteer/puppeteer/files/9958912/puppeteer_multi_page.pdf)
I am not sure if its a bug or I am doing something wrong here.
Puppeteer Code
```
const browser = await puppeteer.launch({
    headless: true,
    args: [
        '--kiosk-printing',
        '--disable-gpu',
        '--disabled-dev-shm-usage',
        '--disabled-setupid-sandbox',
        '--no-first-run',
        '--no-sandbox',
        '--no-zygote',
        '--single-process',
    ],
});
const page = await browser.newPage();
await page.setViewport({ width: 1366, height: 768});
await page.goto(`${req.body.url}`, {waitUntil: ['domcontentloaded', 'networkidle0']});
const pdf = await page.pdf({
    format: 'A4',
    landscape: false,
    printBackground: false,
    preferCSSPageSize: false,
    margin: {
        top: 20,
        bottom: 20,
        left: 20,
        right: 20,
    }
});
     
await browser.close();
```

### Puppeteer version

^19.2.2

### Node.js version

v16.16.0

### npm version

8.11.0

### What operating system are you seeing the problem on?

macOS

### Configuration file

```JavaScript
const {join} = require('path');

/**
 * @type {import("puppeteer").Configuration}
 */
module.exports = {
    // Changes the cache location for Puppeteer.
    cacheDirectory: join(__dirname, '.cache', 'puppeteer'),
};
```

**Comment by OrKoN** [comment]
Do you see the same difference if you run with `headless: false`?

**Comment by zahidhasanemon** [comment]
@OrKoN Yes, the same issue persists with `headless: false`

**Comment by kalakhr** [comment]
I am having the same issue which started with v19.2.0.  v19.1.2 works fine.

**Comment by OrKoN** [comment]
@zahidhasanemon could you provide a test HTML page that reproduces the issue?

**Comment by zahidhasanemon** [comment]
@OrKoN I can confirm that v19.1.2 does not have the issue. This occurs with v19.2.0. I am trying to provide a test html page asap.

**Comment by zahidhasanemon** [comment]
@OrKoN I was unable to reproduce it with raw html. This happens in my vue js application with v19.2.0. If I use v19.1.2, everything works as expected. @kalakhr Can you please provide a test file?

**Comment by kalakhr** [comment]
Please attached files.

[test.txt](https://github.com/puppeteer/puppeteer/files/9982504/test.txt)
[produced-by-v19-1-2.pdf](https://github.com/puppeteer/puppeteer/files/9982507/produced-by-v19-1-2.pdf)
[produced-by-v19-2-0.pdf](https://github.com/puppeteer/puppeteer/files/9982510/produced-by-v19-2-0.pdf)

**Comment by OrKoN** [comment]
Filed the issue upstream as it's the browser issue https://crbug.com/1383166 the only difference between the Puppeteer versions is the version of Chromium that is bundled.

**Comment by kalakhr** [comment]
![image](https://user-images.githubusercontent.com/7253048/201156565-5efe3cb9-a08c-4f67-8f16-ddf67ebff819.png)

**Comment by OrKoN** [comment]
It looks like a regression in Chromium: you should be able to work around it by specifying the following argument when launching puppeteer: `--disable-blink-features=LayoutNGPrinting`

**Comment by dcecile** [comment]
The `--disable-blink-features=LayoutNGPrinting` workaround fixed the issue for me. (Running 19.2.2, had a page incorrectly overflowing.) Thanks @OrKoN!

**Comment by OrKoN** [comment]
This should get fixed with the next release (19.6.0) which includes the latest Chromium binary.
